### PR TITLE
add upload coverage actions

### PR DESCRIPTION
**Pull request by knorth55**

### CODE REVIEW

File: upload_test_coverage/action.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                flags:
+                  - integration
+              unit:
+                flags:
+                  - unit
+              all:
+                flags:
+                  - all
+        fixes:
+          - "/_work/${REPO_NAME}/${REPO_NAME}/src/project::"
```
Comment (furushchev) [inline upload_test_coverage/action.yml:60]: I think we can get full path by using `{{ github.workspace }}/{{ inputs.catkin_workspace }}/src/project`

File: upload_test_coverage/action.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+          - "tests"
+          - ".*"
+        flag_management:
+          default_rules:
+            carryforward: true
+        comment:
+          show_carryforward_flags: true
+        EOF
+      shell: bash
+      working-directory: src/project
```
Comment (furushchev) [inline upload_test_coverage/action.yml:73]: `${{ inputs.catkin_workspace }}/src/project` ?

**Review by furushchev — APPROVED** [review]

File: upload_test_coverage/action.yml (hunk 1 above)
Comment (knorth55) [reply upload_test_coverage/action.yml]: I update to get full path from scripts.

File: upload_test_coverage/action.yml (hunk 2 above)
Comment (knorth55) [reply upload_test_coverage/action.yml:73]: I update to get relateive path

File: upload_test_coverage/action.yml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      id: set_repo_path
+      run: |
+        REPO_NAME=$(echo "${{ github.repository }}" | cut -f 2 -d "/")
+        if [ "${{ inputs.catkin_workspace }}" = "." ]; then
+          REPO_PATH="src/project"
+        else
+          REPO_PATH="${{ inputs.catkin_workspace }}/src/project"
+        fi
+        echo "repo_relative_path=./${REPO_PATH}" >> "$GITHUB_OUTPUT"
+        echo "repo_absolute_path=/_work/${REPO_NAME}/${REPO_NAME}/${REPO_PATH}" >> "$GITHUB_OUTPUT"
```
Comment (furushchev) [inline upload_test_coverage/action.yml:38]: Can we simplify these line with `readlink -f "{{ github.workspace }}/${{ inputs.catkin_workspace }}/src/project" ?

File: upload_test_coverage/action.yml (hunk 1 above)
Comment (knorth55) [reply upload_test_coverage/action.yml]: `${{ github.workspace }}` seems a bit different from `/_work/${REPO_NAME}/${REPO_NAME}`.

File: upload_test_coverage/action.yml (hunk 3 above)
Comment (furushchev) [reply upload_test_coverage/action.yml:38]: Or `readlink -f "${GITHUB_WORKSPACE}/{{ inputs.catkin_workspace }}/src/project"`

Comment (knorth55) [reply upload_test_coverage/action.yml:38]: I'm testing now on `test_ga`.

Comment (knorth55) [reply upload_test_coverage/action.yml]: Thanks! I tested and checked it works!

Comment (furushchev) [reply upload_test_coverage/action.yml]: Cool! Thanks for the check!

**Pull request merged**